### PR TITLE
Disable dark mode in Python docs.

### LINKED ISSUE #14463: [DOC] Higher contrast fonts in dark mode
Currently when I navigate to cudf docs I find that it defaults to dark mode (my OS system defaults) and text is hard to read:

Screenshot 2023-11-21 at 2 11 23 PM

Light mode is great

Screenshot 2023-11-21 at 2 11 36 PM

An easy solution to this would be to just turn off dark mode.  This can be acheived with a setting in conf.py .  I think the setting itself is documented in the pydata sphinx theme y'all are using.

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR disables dark mode in Python docs. The theme is not very legible due to its use of a custom color theme.

Closes #14463.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by bdice** [comment]
Reviewers -- please check the docs preview before merging. I followed the docs at https://pydata-sphinx-theme.readthedocs.io/en/latest/user_guide/light-dark.html to disable this but haven't rendered it locally yet.

**Review by ajschmidt8 — APPROVED** [review]

**Comment by bdice** [comment]
This works... halfway. The button to use dark mode has been removed. However, the webpage still used dark mode if I set my system to dark mode _until_ I cleared my site data for `rapids.ai`. The PyData Sphinx theme uses local storage to store the preferred mode, so my previous "auto" preference was being saved -- I think. After this PR is merged, users may need to clear their cookies / site storage to make the site always light.

-----------------------------

Before:
image

-----------------------------

After (light/dark switcher removed):
image

**Review by galipremsagar — CHANGES_REQUESTED** [review]
Blocking merge temporarily for a simpler approach

**Comment by bdice** [comment]
Closing in favor of https://github.com/rapidsai/docs/pull/456

**Pull request closed without merging**
